boards/xtensa/intel_s1000_crb/intel_s1000_crb.dts

```diff
@@ -35,11 +36,53 @@
 
 &spi0 {
 	status = "ok";
+	#address-cells = <1>;
```

Comment:
NACK: This isn't correct for a SPI node.

The size-cells should be 0.

Reply:
I get a parse error on XCC if i make it 0.

Reply:
> I get a parse error on XCC if i make it 0.

can you report a bug about this? Not sure why this should be XCC specific.

Reply:
Well, checked again and am seeing a different behavior this time. If i keep the address-cells as 1, i get a warning when i do CMake (which i talked about here in https://github.com/zephyrproject-rtos/zephyr/pull/11850#issuecomment-444198345). If i make the address-cells as 0, this warning during CMake goes away but the subsequent "make" fails with the below error.
zephyr/drivers/flash/spi_nor.c:387: error: ‘CONFIG_FLASH_SIZE’ undeclared here (not in a function)

Reply:
how is this related to XCC? looks like something else is wrong here.

Reply:
As per the dts/bindings/slave/spi,flash.yaml you should have size encoded not in reg but as a different property called 'size'.  So make this:

```
#size-cells = <0>;

flash0: flash@0 {
...
    reg = <0>;
    size = <0x400000>;
...
};

Reply:
You will need a fixup for CONFIG_FLASH_SIZE

Reply:
Thanks. Just uploaded with the modifications you suggested.